<?xml version="1.0" encoding="utf-8"?>

<launch>
	<arg name="mav_name" default="tupac" />

	<arg name="result_draw_markers" default="false" />
	<arg name="result_draw_markers_cube" default="false" />
	<arg name="result_draw_markers_axis" default="false" />

	<arg name="uid" default=""/>
	<arg name="video_rectified" default="false" />
	<arg name="use_camera_info" default="true" />
	<arg name="publish_tf" default="true" />

	<arg name="board_config" default="$(find ar_sys)/data/single/single_tag.yml" />
	<arg name="board_frame_id" default="board1" />
	<arg name="marker_size" default="0.20" />

	<!-- ar sys cpp node -->
	<group ns="$(arg mav_name)">
		<node pkg="ar_sys" type="single_board" name="ar_single_board$(arg uid)" output="screen">
			<remap from="/camera_info" to="camera_downward/camera_info" />
			<remap from="/image" to="camera_downward/image_raw" />
			<remap from="ar_single_board/pose" to="tag_pose" />
			<remap from="/camera_info2" to="fake/camera_front/camera_info" />
			<remap from="/image2" to="fake/camera_front/image_raw" />
			<remap from="ar_single_board/pose2" to="fake/tag_pose" />

			<param name="image_is_rectified" type="bool" value="$(arg video_rectified)"/>
			<param name="use_camera_info" type="bool" value="$(arg use_camera_info)"/>
			<param name="board_config" type="string" value="$(arg board_config)"/>
			<param name="board_frame" type="string" value="/$(arg board_frame_id)" />
			<param name="marker_size" type="double" value="$(arg marker_size)"/>
			<param name="draw_markers" type="bool" value="$(arg result_draw_markers)" />
			<param name="draw_markers_cube" type="bool" value="$(arg result_draw_markers_cube)" />
			<param name="draw_markers_axis" type="bool" value="$(arg result_draw_markers_axis)" />
	    <param name="publish_tf" value="$(arg publish_tf)" />
		</node>

		<!--node pkg="topic_tools" type="relay" name="ar_single_board_relay$(arg uid)" args="/ar_single_board$(arg uid)/transform /arsys_single_board/transform" /-->

	</group>
</launch>
